src/if-run/builtins/csv-import/index.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable eqeqeq */
+import {readFile} from 'fs/promises';
+import axios from 'axios';
+import {z} from 'zod';
+import {parse} from 'csv-parse/sync';
+
+import {ConfigParams, PluginParams} from '@grnsft/if-core/types';
+import {PluginFactory} from '@grnsft/if-core/interfaces';
+import {ERRORS, validate} from '@grnsft/if-core/utils';
+
+import {STRINGS} from '../../config';
+
+const {
+  FILE_FETCH_FAILED,
+  FILE_READ_FAILED,
+  MISSING_CSV_COLUMN,
+  MISSING_CONFIG,
+} = STRINGS;
+
+const {
+  FetchingFileError,
+  ReadFileError,
+  MissingCSVColumnError,
+  ConfigError,
+  CSVParseError,
+} = ERRORS;
+
+export const CSVImport = PluginFactory({
+  configValidation: (config: ConfigParams) => {
+    if (!config || !Object.keys(config)?.length) {
+      throw new ConfigError(MISSING_CONFIG);
+    }
+
+    const configSchema = z.object({
+      filepath: z.string(),
+      output: z
+        .string()
+        .or(z.array(z.string()))
+        .or(z.array(z.array(z.string()))),
+    });
+
+    return validate<z.infer<typeof configSchema>>(configSchema, config);
+  },
+  implementation: async (inputs: PluginParams[], config: ConfigParams) => {
+    /**
+     * 1. Tries to retrieve given file (with url or local path).
+     * 2. Parses given CSV.
+     * 3. Filters requested information from CSV.
+     */
+    const {filepath, output} = config;
+    const file = await retrieveFile(filepath);
+    const parsedCSV = parseCSVFile(file);
+
+    const result = parsedCSV?.map((input: PluginParams) =>
+      filterOutput(input, output)
+    );
+
+    return [...inputs, ...result];
+  },
+});
+
+/**
+ * Checks if given string is URL.
+ */
+const isURL = (filepath: string) => {
```

Comment:
@manushak this and some of the other functions are duplicates, let's think of
 optimal way